repo/format/content_format.go

```diff
@@ -63,10 +67,11 @@ func (f *ContentFormat) SupportsPasswordChange() bool {
 // MutableParameters represents parameters of the content manager that can be mutated after the repository
 // is created.
 type MutableParameters struct {
-	Version         Version          `json:"version,omitempty"`         // version number, must be "1", "2" or "3"
-	MaxPackSize     int              `json:"maxPackSize,omitempty"`     // maximum size of a pack object
-	IndexVersion    int              `json:"indexVersion,omitempty"`    // force particular index format version (1,2,..)
-	EpochParameters epoch.Parameters `json:"epochParameters,omitempty"` // epoch manager parameters
+	Version             Version          `json:"version,omitempty"`             // version number, must be "1", "2" or "3"
+	MaxPackSize         int              `json:"maxPackSize,omitempty"`         // maximum size of a pack object
+	IndexVersion        int              `json:"indexVersion,omitempty"`        // force particular index format version (1,2,..)
+	EpochParameters     epoch.Parameters `json:"epochParameters,omitempty"`     // epoch manager parameters
+	MetadataCompression compression.Name `json:"metadataCompression,omitempty"` // metadata compression algorithm name
```

Comment:
The approach implemented in this PR is to have a repo-wide setting in `MutableParameters`.

Does it make sense to make this a policy setting instead?
Are there any tradeoffs in doing so? and which ones?

Reply:
The main disadvantage was the implementation and testing complexity. Since this feature is not expected to be used frequently (applicable only to special cases), we decided to go with the easiest option which will be quicker to implement.